pkg/containers/datasource.go

```diff
@@ -0,0 +1,72 @@
+package containers
+
+import (
+	"encoding/json"
+
+	"github.com/aquasecurity/tracee/types/detect"
+)
+
+type SignaturesDataSource struct {
+	containers *Containers
+}
+
+func NewDataSource(c *Containers) *SignaturesDataSource {
+	return &SignaturesDataSource{
+		containers: c,
+	}
+}
+
+func (ctx SignaturesDataSource) Get(key interface{}) (map[string]interface{}, error) {
+	containerId, ok := key.(string)
+	if !ok {
+		return nil, detect.ErrKeyNotSupported
+	}
+	ctx.containers.mtx.RLock()
+	defer ctx.containers.mtx.RUnlock()
+	for _, cgroup := range ctx.containers.cgroupsMap {
```

Comment:
Thinking out loud: should we worry about this iteration performance ? should we have a map indexed by containerID (and updated during CgroupInfo updates) ? etc...

Reply:
This is an example implementation. I'm not sure if it's optimal performance wise (likely not), I initially toyed with improving it but that led me to playing with the cgroups package internals, which I felt was out of scope. Probably adding caching at the datasource layer could help.

Reply:
Ah gotcha, sorry, did not mean to rush things!

Reply:
I left the same comment in the other PR that was merged.
In the future, we will probably want to add such a map to be O(1) for getting a container.